debug bot exception while trying to register with wonnode

* restart of bot seems to solve the connection problem => maybe bot starts up before won node is there and cannot properly connect then?
* check also other bots for this bug

``
20.1.17 16:41:59.742 [Thread-12] INFO  w.o.m.OwnerWonMessageSenderJMSBased - register at default won node https://node.matchat.org/won/resource
20.1.17 16:41:59.834 [Thread-11] INFO  w.o.m.OwnerProtocolCommunicationServiceImpl - we're not yet registered. Registering with WoN node:https://node.matchat.org/won/resource
20.1.17 16:42:00.095 [Thread-11] INFO  won.cryptography.ssl.TOFUStrategy - Certificate for 
4956f0246efe526c3230bc6908118da886c6fd1c70f6e754e9974ddd is added based on TOFU and from now on it is trusted!
20.1.17 16:42:09.693 [Thread-14] INFO  w.o.m.OwnerWonMessageSenderJMSBased - register at default won node https://node.matchat.org/won/resource
20.1.17 16:42:10.171 [Thread-11] WARN  w.o.m.OwnerWonMessageSenderJMSBased - Could not register with default won node https://node.matchat.org/won/resource. Try again later ... (reason is logged at loglevel 'DEBUG')
20.1.17 16:42:10.172 [Thread-14] INFO  w.o.m.OwnerProtocolCommunicationServiceImpl - setting up communication with won node: https://node.matchat.org/won/resource
20.1.17 16:42:10.185 [Thread-14] INFO  w.o.m.OwnerProtocolCommunicationServiceImpl - we're not yet registered. Registering with WoN node:https://node.matchat.org/won/resource
20.1.17 16:42:19.692 [Thread-15] INFO  w.o.m.OwnerWonMessageSenderJMSBased - register at default won node https://node.matchat.org/won/resource
20.1.17 16:42:20.242 [Thread-12] INFO  w.o.m.OwnerProtocolCommunicationServiceImpl - setting up communication with won node: https://node.matchat.org/won/resource
20.1.17 16:42:20.242 [Thread-14] WARN  w.o.m.OwnerWonMessageSenderJMSBased - Could not register with default won node https://node.matchat.org/won/resource. Try again later ... (reason is logged at loglevel 'DEBUG')
20.1.17 16:42:20.245 [Thread-12] INFO  w.o.m.OwnerProtocolCommunicationServiceImpl - we're not yet registered. Registering with WoN node:https://node.matchat.org/won/resource
20.1.17 16:42:27.227 [Thread-13] WARN  w.m.p.i.MatcherProtocolMatcherServiceImplJMSBased - Could not get topic lists from won node https://node.matchat.org/won/resource
java.lang.IllegalArgumentException: caught a clientHandler exception, which may indicate that the URI that was accessed isn't a linked data URI, please check https://node.matchat.org/won/resource. Underlying error message is: I/O error on GET request for "https://node.matchat.org/won/resource":Read timed out; nested exception is java.net.SocketTimeoutException: Read timed out
        at won.protocol.rest.LinkedDataRestClient.readResourceData(LinkedDataRestClient.java:118) ~[bots.jar:na]
        at won.protocol.rest.LinkedDataRestClientHttps.readResourceDataWithHeaders(LinkedDataRestClientHttps.java:134) ~[bots.jar:na]
        at won.protocol.util.linkeddata.CachingLinkedDataSource.fetch(CachingLinkedDataSource.java:341) ~[bots.jar:na]
        at won.protocol.util.linkeddata.CachingLinkedDataSource.fetchWithEtagValidation(CachingLinkedDataSource.java:298) ~[bots.jar:na]
        at won.protocol.util.linkeddata.CachingLinkedDataSource.fetchAndCacheIfAppropriate(CachingLinkedDataSource.java:235) ~[bots.jar:na]
        at won.protocol.util.linkeddata.CachingLinkedDataSource.fetchOnlyOnce(CachingLinkedDataSource.java:217) ~[bots.jar:na]
        at won.protocol.util.linkeddata.CachingLinkedDataSource.fetchOrUseCached(CachingLinkedDataSource.java:164) ~[bots.jar:na]
        at won.protocol.util.linkeddata.CachingLinkedDataSource.getDataForResource(CachingLinkedDataSource.java:108) ~[bots.jar:na]
        at won.protocol.util.linkeddata.CachingLinkedDataSource.getDataForResource(CachingLinkedDataSource.java:429) ~[bots.jar:na]
        at won.protocol.jms.MatcherActiveMQServiceImpl.getMatcherProtocolTopicNamesWithResource(MatcherActiveMQServiceImpl.java:71) ~[bots.jar:na]
        at won.matcher.protocol.impl.MatcherProtocolCommunicationServiceImpl.getMatcherProtocolOutTopics(MatcherProtocolCommunicationServiceImpl.java:81) ~[bots.jar:na]
        at won.matcher.protocol.impl.MatcherProtocolMatcherServiceImplJMSBased.configureMatcherProtocolOutTopics(MatcherProtocolMatcherServiceImplJMSBased.java:81) ~[bots.jar:na]
        at won.matcher.protocol.impl.MatcherProtocolMatcherServiceImplJMSBased.access$000(MatcherProtocolMatcherServiceImplJMSBased.java:43) ~[bots.jar:na]
        at won.matcher.protocol.impl.MatcherProtocolMatcherServiceImplJMSBased$1.run(MatcherProtocolMatcherServiceImplJMSBased.java:95) ~[bots.jar:na]
Caused by: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "https://node.matchat.org/won/resource":Read timed out; nested exception is java.net.SocketTimeoutException: Read timed out
        at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:584) ~[bots.jar:na]
        at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:544) ~[bots.jar:na]
        at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:465) ~[bots.jar:na]
        at won.protocol.rest.LinkedDataRestClient.readResourceData(LinkedDataRestClient.java:100) ~[bots.jar:na]
        ... 13 common frames omitted
Caused by: java.net.SocketTimeoutException: Read timed out
        at java.net.SocketInputStream.socketRead0(Native Method) ~[na:1.8.0_111]
        at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[na:1.8.0_111]
        at java.net.SocketInputStream.read(SocketInputStream.java:170) ~[na:1.8.0_111]
        at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[na:1.8.0_111]
        at sun.security.ssl.InputRecord.readFully(InputRecord.java:465) ~[na:1.8.0_111]
        at sun.security.ssl.InputRecord.read(InputRecord.java:503) ~[na:1.8.0_111]
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973) ~[na:1.8.0_111]
        at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:930) ~[na:1.8.0_111]
        at sun.security.ssl.AppInputStream.read(AppInputStream.java:105) ~[na:1.8.0_111]
        at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:139) ~[bots.jar:na]
        at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:155) ~[bots.jar:na]
        at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:284) ~[bots.jar:na]
        at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:140) ~[bots.jar:na]
        at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:57) ~[bots.jar:na]
        at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:261) ~[bots.jar:na]
        at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:165) ~[bots.jar:na]
        at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:167) ~[bots.jar:na]
        at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:272) ~[bots.jar:na]
        at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:124) ~[bots.jar:na]
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:271) ~[bots.jar:na]
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184) ~[bots.jar:na]
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88) ~[bots.jar:na]
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) ~[bots.jar:na]
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184) ~[bots.jar:na]
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82) ~[bots.jar:na]
        at org.springframework.http.client.HttpComponentsClientHttpRequest.executeInternal(HttpComponentsClientHttpRequest.java:91) ~[bots.jar:na]
        at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48) ~[bots.jar:na]
        at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:53) ~[bots.jar:na]
        at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:568) ~[bots.jar:na]
        ... 16 common frames omitted
``

--- a/webofneeds/won-bot/src/main/java/won/bot/framework/eventbot/action/impl/matcher/RegisterMatcherAction.java
+++ b/webofneeds/won-bot/src/main/java/won/bot/framework/eventbot/action/impl/matcher/RegisterMatcherAction.java
@@ -20,9 +20,12 @@ import won.bot.framework.eventbot.EventListenerContext;
 import won.bot.framework.eventbot.action.BaseEventBotAction;
 import won.bot.framework.eventbot.action.EventBotActionUtils;
 import won.bot.framework.eventbot.event.Event;
+import won.bot.framework.eventbot.event.impl.matcher.MatcherRegisterFailedEvent;
 
 import java.net.URI;
 import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
 
 /**
 * User: fkleedorfer
@@ -30,22 +33,31 @@ import java.util.Iterator;
 */
 public class RegisterMatcherAction extends BaseEventBotAction
 {
+  private List<URI> registeredNodes = new LinkedList<>();
+
   public RegisterMatcherAction(final EventListenerContext eventListenerContext)
   {
     super(eventListenerContext);
   }
 
-
-    @Override
+  @Override
   protected void doRun(Event event) throws Exception
   {
     final Iterator wonNodeUriIterator = getEventListenerContext().getMatcherNodeURISource().getNodeURIIterator();
     while (wonNodeUriIterator.hasNext()){
       URI wonNodeUri = (URI)wonNodeUriIterator.next();
-      logger.debug("registering matcher on won node {}", wonNodeUri);
-      getEventListenerContext().getMatcherProtocolMatcherService().register( wonNodeUri );
-      EventBotActionUtils.rememberInNodeListIfNamePresent(getEventListenerContext(),wonNodeUri);
-      logger.debug("matcher registered on won node {}", wonNodeUri);
+      try {
+        if (!registeredNodes.contains(wonNodeUri)) {
+          logger.debug("registering matcher on won node {}", wonNodeUri);
+          EventBotActionUtils.rememberInNodeListIfNamePresent(getEventListenerContext(), wonNodeUri);
+          getEventListenerContext().getMatcherProtocolMatcherService().register(wonNodeUri);
+          registeredNodes.add(wonNodeUri);
+          logger.info("matcher registered on won node {}", wonNodeUri);
+        }
+      } catch (Exception e) {
+        logger.warn("Error registering matcher at won node {}. Try again later ... Exception was {}", wonNodeUri, e);
+        getEventListenerContext().getEventBus().publish(new MatcherRegisterFailedEvent(wonNodeUri));
+      }
     }
   }
 

--- a/webofneeds/won-bot/src/main/java/won/bot/framework/eventbot/event/impl/matcher/MatcherRegisterFailedEvent.java
+++ b/webofneeds/won-bot/src/main/java/won/bot/framework/eventbot/event/impl/matcher/MatcherRegisterFailedEvent.java
@@ -1,0 +1,15 @@
+package won.bot.framework.eventbot.event.impl.matcher;
+
+import won.bot.framework.eventbot.event.BaseNodeSpecificEvent;
+
+import java.net.URI;
+
+/**
+ * Created by hfriedrich on 26.01.2017.
+ */
+public class MatcherRegisterFailedEvent extends BaseNodeSpecificEvent
+{
+  public MatcherRegisterFailedEvent(final URI wonNodeURI) {
+    super(wonNodeURI);
+  }
+}

--- a/webofneeds/won-bot/src/main/java/won/bot/impl/DebugBot.java
+++ b/webofneeds/won-bot/src/main/java/won/bot/impl/DebugBot.java
@@ -18,7 +18,8 @@ package won.bot.impl;
 
 import won.bot.framework.bot.base.EventBot;
 import won.bot.framework.eventbot.EventListenerContext;
-import won.bot.framework.eventbot.action.impl.*;
+import won.bot.framework.eventbot.action.impl.MultipleActions;
+import won.bot.framework.eventbot.action.impl.RandomDelayedAction;
 import won.bot.framework.eventbot.action.impl.debugbot.*;
 import won.bot.framework.eventbot.action.impl.matcher.RegisterMatcherAction;
 import won.bot.framework.eventbot.action.impl.needlifecycle.DeactivateNeedAction;
@@ -27,6 +28,7 @@ import won.bot.framework.eventbot.bus.EventBus;
 import won.bot.framework.eventbot.event.impl.command.SendTextMessageOnConnectionEvent;
 import won.bot.framework.eventbot.event.impl.debugbot.*;
 import won.bot.framework.eventbot.event.impl.lifecycle.ActEvent;
+import won.bot.framework.eventbot.event.impl.matcher.MatcherRegisterFailedEvent;
 import won.bot.framework.eventbot.event.impl.matcher.NeedCreatedEventForMatcher;
 import won.bot.framework.eventbot.event.impl.wonmessage.*;
 import won.bot.framework.eventbot.filter.impl.NeedUriInNamedListFilter;
@@ -58,6 +60,11 @@ public class DebugBot extends EventBot
   protected BaseEventListener needDeactivator;
   protected BaseEventListener messageFromOtherNeedListener;
   protected BaseEventListener usageMessageSender;
+  private int registrationMatcherRetryInterval;
+
+  public void setRegistrationMatcherRetryInterval(final int registrationMatcherRetryInterval) {
+    this.registrationMatcherRetryInterval = registrationMatcherRetryInterval;
+  }
 
 
   private URI matcherUri;
@@ -76,13 +83,13 @@ public class DebugBot extends EventBot
     EventListenerContext ctx = getEventListenerContext();
     EventBus bus = getEventBus();
 
-    //register with WoN noodes, be notified when new needs are created
-    this.matcherRegistrator = new ActionOnEventListener(
-      ctx,
-      new RegisterMatcherAction(ctx),
-      1
-    );
+    //register with WoN nodes, be notified when new needs are created
+    RegisterMatcherAction registerMatcherAction = new RegisterMatcherAction(ctx);
+    this.matcherRegistrator = new ActionOnEventListener(ctx, registerMatcherAction, 1);
     bus.subscribe(ActEvent.class, this.matcherRegistrator);
+    RandomDelayedAction delayedRegistration = new RandomDelayedAction(ctx, registrationMatcherRetryInterval, registrationMatcherRetryInterval, 0, registerMatcherAction);
+    ActionOnEventListener matcherRetryRegistrator = new ActionOnEventListener(ctx, delayedRegistration);
+    bus.subscribe(MatcherRegisterFailedEvent.class, matcherRetryRegistrator);
 
     //create the echo need for debug initial connect - if we're not reacting to the creation of our own echo need.
     CreateDebugNeedWithFacetsAction needForInitialConnectAction =

--- a/webofneeds/won-bot/src/main/java/won/bot/impl/EchoBot.java
+++ b/webofneeds/won-bot/src/main/java/won/bot/impl/EchoBot.java
@@ -26,6 +26,7 @@ import won.bot.framework.eventbot.action.impl.wonmessage.ConnectWithAssociatedNe
 import won.bot.framework.eventbot.action.impl.wonmessage.RespondWithEchoToMessageAction;
 import won.bot.framework.eventbot.bus.EventBus;
 import won.bot.framework.eventbot.event.impl.lifecycle.ActEvent;
+import won.bot.framework.eventbot.event.impl.matcher.MatcherRegisterFailedEvent;
 import won.bot.framework.eventbot.event.impl.matcher.NeedCreatedEventForMatcher;
 import won.bot.framework.eventbot.event.impl.needlifecycle.NeedCreatedEvent;
 import won.bot.framework.eventbot.event.impl.wonmessage.CloseFromOtherNeedEvent;
@@ -52,8 +53,12 @@ public class EchoBot extends EventBot
   protected BaseEventListener autoResponder;
   protected BaseEventListener connectionCloser;
   protected BaseEventListener needDeactivator;
-
   private Integer numberOfEchoNeedsPerNeed;
+  private int registrationMatcherRetryInterval;
+
+  public void setRegistrationMatcherRetryInterval(final int registrationMatcherRetryInterval) {
+    this.registrationMatcherRetryInterval = registrationMatcherRetryInterval;
+  }
 
   @Override
   protected void initializeEventListeners()
@@ -61,13 +66,13 @@ public class EchoBot extends EventBot
     EventListenerContext ctx = getEventListenerContext();
     EventBus bus = getEventBus();
 
-    //register with WoN noodes, be notified when new needs are created
-    this.matcherRegistrator = new ActionOnEventListener(
-      ctx,
-      new RegisterMatcherAction(ctx),
-      1
-    );
+    //register with WoN nodes, be notified when new needs are created
+    RegisterMatcherAction registerMatcherAction = new RegisterMatcherAction(ctx);
+    this.matcherRegistrator = new ActionOnEventListener(ctx, registerMatcherAction, 1);
     bus.subscribe(ActEvent.class, this.matcherRegistrator);
+    RandomDelayedAction delayedRegistration = new RandomDelayedAction(ctx, registrationMatcherRetryInterval, registrationMatcherRetryInterval, 0, registerMatcherAction);
+    ActionOnEventListener matcherRetryRegistrator = new ActionOnEventListener(ctx, delayedRegistration);
+    bus.subscribe(MatcherRegisterFailedEvent.class, matcherRetryRegistrator);
 
     //create the echo need - if we're not reacting to the creation of our own echo need.
     this.needCreator = new ActionOnEventListener(

--- a/webofneeds/won-bot/src/main/java/won/bot/impl/LastSeenNeedsMatcherBot.java
+++ b/webofneeds/won-bot/src/main/java/won/bot/impl/LastSeenNeedsMatcherBot.java
@@ -19,10 +19,12 @@ package won.bot.impl;
 import won.bot.framework.bot.base.EventBot;
 import won.bot.framework.eventbot.EventListenerContext;
 import won.bot.framework.eventbot.action.BaseEventBotAction;
+import won.bot.framework.eventbot.action.impl.RandomDelayedAction;
 import won.bot.framework.eventbot.action.impl.matcher.RegisterMatcherAction;
 import won.bot.framework.eventbot.bus.EventBus;
 import won.bot.framework.eventbot.event.Event;
 import won.bot.framework.eventbot.event.impl.lifecycle.ActEvent;
+import won.bot.framework.eventbot.event.impl.matcher.MatcherRegisterFailedEvent;
 import won.bot.framework.eventbot.event.impl.matcher.NeedCreatedEventForMatcher;
 import won.bot.framework.eventbot.listener.BaseEventListener;
 import won.bot.framework.eventbot.listener.impl.ActionOnEventListener;
@@ -44,6 +46,11 @@ public class LastSeenNeedsMatcherBot extends EventBot
 
   private BaseEventListener matcherRegistrator;
   private BaseEventListener matcherIndexer;
+  private int registrationMatcherRetryInterval;
+
+  public void setRegistrationMatcherRetryInterval(final int registrationMatcherRetryInterval) {
+    this.registrationMatcherRetryInterval = registrationMatcherRetryInterval;
+  }
 
   private URI matcherUri;
 
@@ -60,12 +67,12 @@ public class LastSeenNeedsMatcherBot extends EventBot
     EventBus bus = getEventBus();
 
     //subscribe this bot with the WoN nodes' 'new need' topic
-    this.matcherRegistrator = new ActionOnEventListener(
-      ctx,
-      new RegisterMatcherAction(ctx),
-      1
-    );
-    bus.subscribe(ActEvent.class,this.matcherRegistrator);
+    RegisterMatcherAction registerMatcherAction = new RegisterMatcherAction(ctx);
+    this.matcherRegistrator = new ActionOnEventListener(ctx, registerMatcherAction, 1);
+    bus.subscribe(ActEvent.class, this.matcherRegistrator);
+    RandomDelayedAction delayedRegistration = new RandomDelayedAction(ctx, registrationMatcherRetryInterval, registrationMatcherRetryInterval, 0, registerMatcherAction);
+    ActionOnEventListener matcherRetryRegistrator = new ActionOnEventListener(ctx, delayedRegistration);
+    bus.subscribe(MatcherRegisterFailedEvent.class, matcherRetryRegistrator);
 
     bus.subscribe(NeedCreatedEventForMatcher.class,
           new ActionOnEventListener(ctx, "lastSeenNeedsMatcher",

--- a/webofneeds/won-bot/src/main/java/won/bot/impl/MatcherProtocolTestBot.java
+++ b/webofneeds/won-bot/src/main/java/won/bot/impl/MatcherProtocolTestBot.java
@@ -28,6 +28,7 @@ import won.bot.framework.eventbot.action.impl.needlifecycle.DeactivateAllNeedsAc
 import won.bot.framework.eventbot.bus.EventBus;
 import won.bot.framework.eventbot.event.Event;
 import won.bot.framework.eventbot.event.impl.lifecycle.ActEvent;
+import won.bot.framework.eventbot.event.impl.matcher.MatcherRegisterFailedEvent;
 import won.bot.framework.eventbot.event.impl.matcher.MatcherRegisteredEvent;
 import won.bot.framework.eventbot.event.impl.matcher.NeedCreatedEventForMatcher;
 import won.bot.framework.eventbot.event.impl.matcher.NeedDeactivatedEventForMatcher;
@@ -58,6 +59,12 @@ public class MatcherProtocolTestBot extends EventBot
   protected BaseEventListener allNeedsDeactivator;
   protected BaseEventListener workDoneSignaller;
   protected BaseEventListener matcherNotifier;
+  private int registrationMatcherRetryInterval;
+
+  public void setRegistrationMatcherRetryInterval(final int registrationMatcherRetryInterval) {
+    this.registrationMatcherRetryInterval = registrationMatcherRetryInterval;
+  }
+
   private static final String NAME_NEEDS = "needs";
 
   @Override
@@ -66,12 +73,14 @@ public class MatcherProtocolTestBot extends EventBot
     EventListenerContext ctx = getEventListenerContext();
     EventBus bus = getEventBus();
 
-    this.matcherRegistrator = new ActionOnEventListener(
-      ctx,
-      new RegisterMatcherAction(ctx),
-      1
-    );
+    //subscribe this bot with the WoN nodes' 'new need' topic
+    RegisterMatcherAction registerMatcherAction = new RegisterMatcherAction(ctx);
+    this.matcherRegistrator = new ActionOnEventListener(ctx, registerMatcherAction, 1);
     bus.subscribe(ActEvent.class, this.matcherRegistrator);
+    RandomDelayedAction delayedRegistration = new RandomDelayedAction(
+      ctx, registrationMatcherRetryInterval, registrationMatcherRetryInterval, 0, registerMatcherAction);
+    ActionOnEventListener matcherRetryRegistrator = new ActionOnEventListener(ctx, delayedRegistration);
+    bus.subscribe(MatcherRegisterFailedEvent.class, matcherRetryRegistrator);
 
     //create needs every trigger execution until 2 needs are created
     this.needCreator = new ActionOnEventListener(

--- a/webofneeds/won-bot/src/main/resources/spring/bot/DebugBot.xml
+++ b/webofneeds/won-bot/src/main/resources/spring/bot/DebugBot.xml
@@ -33,6 +33,8 @@
         <!--property name="ownerService" ref="ownerProtocolNeedServiceClient" /-->
         <property name="matcherProtocolNeedServiceClient" ref="matcherProtocolNeedServiceClient"/>
         <property name="matcherProtocolMatcherService" ref="matcherProtocolMatcherServiceJMSBased"/>
+        <property name="registrationMatcherRetryInterval" value="5000"/>
+        
         <property name="trigger">
             <bean class="org.springframework.scheduling.support.PeriodicTrigger">
                 <constructor-arg name="period" value="2000" />

--- a/webofneeds/won-bot/src/main/resources/spring/bot/EchoBot.xml
+++ b/webofneeds/won-bot/src/main/resources/spring/bot/EchoBot.xml
@@ -33,6 +33,8 @@
         <!--property name="ownerService" ref="ownerProtocolNeedServiceClient" /-->
         <property name="matcherProtocolNeedServiceClient" ref="matcherProtocolNeedServiceClient"/>
         <property name="matcherProtocolMatcherService" ref="matcherProtocolMatcherServiceJMSBased"/>
+        <property name="registrationMatcherRetryInterval" value="5000"/>
+
         <property name="trigger">
             <bean class="org.springframework.scheduling.support.PeriodicTrigger">
                 <constructor-arg name="period" value="2000" />

--- a/webofneeds/won-bot/src/main/resources/spring/bot/LastSeenNeedsMatcherBot.xml
+++ b/webofneeds/won-bot/src/main/resources/spring/bot/LastSeenNeedsMatcherBot.xml
@@ -33,6 +33,8 @@
         <!--property name="nodeURISource" ref="nodeUriSourceRandom" /-->
         <property name="matcherNodeURISource" ref="matcherNodeURISource"/>
         <property name="matcherUri" value="${matcher.uri}" />
+        <property name="registrationMatcherRetryInterval" value="5000"/>
+
         <property name="trigger">
             <bean class="org.springframework.scheduling.support.PeriodicTrigger">
                 <constructor-arg name="period" value="2000" />

--- a/webofneeds/won-bot/src/main/resources/spring/bot/mailBasedSimpleReactiveBotMatcher.xml
+++ b/webofneeds/won-bot/src/main/resources/spring/bot/mailBasedSimpleReactiveBotMatcher.xml
@@ -33,6 +33,8 @@
         <property name="wonMessageSender" ref="ownerProtocolNeedServiceClientJMSBased" />
         <property name="matcherProtocolNeedServiceClient" ref="matcherProtocolNeedServiceClient"/>
         <property name="matcherProtocolMatcherService" ref="matcherProtocolMatcherServiceJMSBased"/>
+        <property name="registrationMatcherRetryInterval" value="5000"/>
+
         <property name="trigger">
             <bean class="org.springframework.scheduling.support.PeriodicTrigger">
                 <constructor-arg name="period" value="2000" />

--- a/webofneeds/won-matcher/src/main/java/won/matcher/protocol/impl/MatcherProtocolMatcherServiceImplJMSBased.java
+++ b/webofneeds/won-matcher/src/main/java/won/matcher/protocol/impl/MatcherProtocolMatcherServiceImplJMSBased.java
@@ -86,23 +86,10 @@ public class MatcherProtocolMatcherServiceImplJMSBased
 
 
 
-  public void register(final URI wonNodeURI) {
-      try {
-        new Thread(){
-          @Override
-          public void run() {
-            try {
-                 configureMatcherProtocolOutTopics(wonNodeURI);
-            } catch (Exception e) {
-              logger.warn("Could not get topic lists from won node {}", wonNodeURI,e);
-            }
-          }
-        }.start();
-      } catch (Exception e) {
-        logger.warn("getting topic lists from the node {} failed",wonNodeURI);
-      }
-
+  public void register(final URI wonNodeURI) throws CamelConfigurationFailedException {
+    configureMatcherProtocolOutTopics(wonNodeURI);
   }
+
   public void register(){
       logger.debug("registering owner application on application event");
       try {

